internal/imex/imex.go

```diff
@@ -0,0 +1,116 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package imex
+
+import (
+	"os"
+	"path/filepath"
+
+	spec "github.com/NVIDIA/k8s-device-plugin/api/config/v1"
+)
+
+// Channels represents a set of IMEX channels.
+type Channels []*Channel
```

Comment:
I'm not sure how useful it is to support more than one channel. I know we originally had NVIDIA_IMEX_CHANNELS take a list, but that was because we didn't really understand how RM treats the available channels it sees. It only picks up and uses the lowest numbered one it sees -- all others are ignored.

Reply:
I don't think we need to support multiple channels, but dealing with this as a slice allows for callers to not have to check for empty values on return. In practice we always iterate over zero or 1 elements.

Let me take another pass at it to see whether using a scalar here would make sense.

Reply:
We can keep the data structure as a slice if that's easier (and even accept the input as a slie with the only valid value being `[0]`).